components/ModEditSubmissionForm.vue

```diff
@@ -692,61 +745,133 @@ const submissionHasUnsavedChanges = () => {
-                    phone: submissionFormFields.phone.value || '',
-                    website: submissionFormFields.website.value || '',
+                    googleMapsUrl: submissionFormFields.googlemapsURL.value,
+                    email: submissionFormFields.email.value,
+                    phone: submissionFormFields.phone.value,
+                    website: submissionFormFields.website.value,
                     address: {
-                        postalCode: submissionFormFields.postalCode.value || '',
-                        prefectureEn: submissionFormFields.prefectureEn.value || '',
-                        cityEn: submissionFormFields.cityEn.value || '',
-                        addressLine1En: submissionFormFields.addressLine1En.value || '',
-                        addressLine2En: submissionFormFields.addressLine2En.value || '',
-                        prefectureJa: submissionFormFields.prefectureJa.value || '',
-                        cityJa: submissionFormFields.cityJa.value || '',
-                        addressLine1Ja: submissionFormFields.addressLine1Ja.value || '',
-                        addressLine2Ja: submissionFormFields.addressLine2Ja.value || ''
+                        postalCode: submissionFormFields.postalCode.value,
+                        prefectureEn: submissionFormFields.prefectureEn.value,
+                        cityEn: submissionFormFields.cityEn.value,
+                        addressLine1En: submissionFormFields.addressLine1En.value,
+                        addressLine2En: submissionFormFields.addressLine2En.value,
+                        prefectureJa: submissionFormFields.prefectureJa.value,
+                        cityJa: submissionFormFields.cityJa.value,
+                        addressLine1Ja: submissionFormFields.addressLine1Ja.value,
+                        addressLine2Ja: submissionFormFields.addressLine2Ja.value
                     }
                 },
                 healthcareProfessionalIds: [],
                 mapLatitude: parseFloat(submissionFormFields.mapLatitude.value) || 0,
                 mapLongitude: parseFloat(submissionFormFields.mapLongitude.value) || 0
-            }
+            },
+            healthcareProfessionals: [
+                {
+                    acceptedInsurance: submissionFormFields.healthcareProfessionalAcceptedInsurances.value,
+                    degrees: submissionFormFields.healthcareProfessionalDegrees.value,
+                    specialties: submissionFormFields.healthcareProfessionalSpecialties.value,
+                    spokenLanguages: submissionFormFields.healthcareProfessionalLocales.value,
+                    names: submissionFormFields.healthCareProfessionalNameArray.value,
+                    facilityIds: []
+                }
+            ]
         }
     }
+    try {
+        const result = await moderationSubmissionStore.updateSubmission(submissionInputVariables)
```

Comment:
When we call `updateSubmission`, I believe it will return the object with the valid `facilityIds` array and `healthcareProfessionalIds` array populated. We should update the object we're using here as I think it might try to wipe the relationships if you hit update again with an empty array.
While I'm writing this, I can see how this wouldn't have been an intuitive logic to pick up on....I'll have to think how we can improve this 🤔

Reply:
Am I not updating with the response in the next line with the `initialize` function?

Reply:
oof, I must have missed that call. That should work 👍 
Let's add a comment to explain that you're updating the form with the values from the api that has updated ids